packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts

```diff
@@ -65,27 +66,39 @@ export function useMessageBarReflow(enabled: boolean = false) {
     [forceUpdate],
   );
 
-  const ref = React.useCallback(
-    (el: HTMLElement | null) => {
+  const ref = React.useMemo(() => {
+    let resizeObserver: ResizeObserver | null = null;
+    return (el: HTMLElement | null) => {
       if (!enabled || !el || !targetDocument?.defaultView) {
+        resizeObserver?.disconnect();
         return;
       }
 
-      resizeObserverRef.current?.disconnect();
+      resizeObserver?.disconnect();
 
       const win = targetDocument.defaultView;
-      const resizeObserver = new win.ResizeObserver(handleResize);
-      resizeObserverRef.current = resizeObserver;
+      resizeObserver = new win.ResizeObserver(handleResize);
       resizeObserver.observe(el, { box: 'border-box' });
-    },
-    [targetDocument, handleResize, enabled],
-  );
-
-  React.useEffect(() => {
```

Comment:
Removed the pattern of creating the resize observer in a callback ref and disposing it in an effect - it is not strict mode compliant actually.

Now the resize observer is created and disposed in the callback ref